appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/impl/TimerImpl.java

```diff
@@ -205,6 +223,52 @@ public String toString() {
         return "Timer[" + getCount() + "]";
     }
 
+    void validateMetricsConfiguration(String metricName, Map<String, Collection<MetricsCustomPercentiles>> percentilesConfigMap,
+                                      Map<String, Collection<MetricsCustomBuckets>> timerBucketsConfigMap) {
+        Collection<MetricsCustomPercentiles> computedPercentiles = percentilesConfigMap
+                .computeIfAbsent(metricName, MetricsConfigParserUtil::processPercentileMap);
+        Collection<MetricsCustomBuckets> computedTimersBuckets = timerBucketsConfigMap
+                .computeIfAbsent(metricName, this::processTimerBucketMap);
+        MetricsCustomPercentiles resultPercentile = null;
+        MetricsCustomBuckets resultBucket = null;
+        configurationProperties = new ConfigurationProperties();
+
+        if (computedPercentiles != null && !computedPercentiles.isEmpty()) {
+            resultPercentile = MetricsCustomPercentiles.matches(computedPercentiles, metricName);
+        }
+
+        if (resultPercentile != null && resultPercentile.getPercentiles() != null
+                && resultPercentile.getPercentiles().length > 0) {
+            configurationProperties.setPercentilesFromConfig(resultPercentile.getPercentiles());
+        } else if (resultPercentile != null && resultPercentile.getPercentiles() == null
+                && resultPercentile.isDisabled()) {
+
+        } else {
+            Double[] percentiles = {0.5, 0.75, 0.95, 0.98, 0.99, 0.999};
+            configurationProperties.setPercentilesFromConfig(percentiles);
+        }
+
+        if (computedTimersBuckets != null && computedTimersBuckets.size() != 0) {
+            resultBucket = MetricsCustomBuckets.matches(computedTimersBuckets, metricName);
+        }
+
+        if (resultBucket != null && resultBucket.getBuckets() != null && resultBucket.getBuckets().length > 0) {
+            configurationProperties.setBucketValuesFromConfig(resultBucket.getBuckets());
+        }
+
+        this.reservoir.setConfigAdapter(configurationProperties);
+    }
+
+    private synchronized Collection<MetricsCustomBuckets> processTimerBucketMap(String appName) {
```

Comment:
Does this _need_ to be synchronized?

Is it an issue if there's some racing with resolving the config value - would it actually change that frequently?
Or is there some server startup or application loading race?